dotnet/src/VectorDataIntegrationTests/MongoDBIntegrationTests/Support/MongoDBTestStore.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Connectors.MongoDB;
+using MongoDB.Driver;
+using Testcontainers.MongoDb;
+using VectorDataSpecificationTests.Support;
+
+namespace MongoDBIntegrationTests.Support;
+
+internal sealed class MongoDBTestStore : TestStore
+{
+    public static MongoDBTestStore Instance { get; } = new();
+
+    private readonly MongoDbContainer _container = new MongoDbBuilder()
+        .WithImage("mongodb/mongodb-atlas-local:7.0.6")
+        .Build();
+
+    public MongoClient? _client { get; private set; }
+    public IMongoDatabase? _database { get; private set; }
+    private MongoDBVectorStore? _defaultVectorStore;
+
+    public MongoClient Client => this._client ?? throw new InvalidOperationException("Not initialized");
+    public IMongoDatabase Database => this._database ?? throw new InvalidOperationException("Not initialized");
+
+    public override MongoDBVectorStore DefaultVectorStore => this._defaultVectorStore ?? throw new InvalidOperationException("Not initialized");
+
+    public MongoDBVectorStore GetVectorStore(MongoDBVectorStoreOptions options)
+        => new(this.Database, options);
+
+    private MongoDBTestStore()
+    {
+    }
+
+    protected override async Task StartAsync()
+    {
+        await this._container.StartAsync();
+
+        this._client = new MongoClient(new MongoClientSettings
+        {
+            Server = new MongoServerAddress(this._container.Hostname, this._container.GetMappedPublicPort(MongoDbBuilder.MongoDbPort)),
```

Comment:
This is very simple, yet powerful. I am glad to see `Testcontainers` being used, so we no longer need to worry about configuring docker in yaml/json and excluding the test credentials from various tools that search for leaked credentials. 🥇

Reply:
Oh yes. One additional great advantage of testcontainers, is that a testcontainer is also responsible for waiting until the service inside is ready - the precise details of how to do this are very service-specific, and that's encapsulated in the testcontainer package as well.